api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -161,6 +161,13 @@
      */
     Attribute<Class<?>> TUPLE_CLASS = new Attribute<>(new Class2String<>());
 
+    /**
+     * Attribute of input port.
+     * This is a read-only attribute to query whether the input port is connected to a DelayOperator
+     * This is for iterative processing.
+     */
+    Attribute<Boolean> IS_CONNECTED_TO_DELAY_OPERATOR = new Attribute<>(false);
```

Comment:
@davidyan74 : I checked the implementation of InputPortMeta.getValue() and looks like you don't need to check for the null..

I made changes in my local branch (https://github.com/gauravgopi123/incubator-apex-core/commit/9f4cf5cd3e4895c268c1d76f436e4db2223bfd5d) and ran tests and it worked fine

Does that work for you?